src/why3_tactic.ml

```diff
@@ -0,0 +1,293 @@
+(** Calling a prover using Why3. *)
+
+open Terms
+open Extra
+open Timed
+open Scope
+
+exception NoGoalTranslation
+
+(* A map from lambdapi terms to Why3 constants. *)
+type cnst_table = (term * Why3.Term.lsymbol) list
+
+(** [get_newname ()] generates a new axiom name. *)
+let get_newname : unit -> string =
+  (* Number of axioms proved whith the Why3 tactic. *)
+  let nbr_axioms : int ref = ref 0 in fun () ->
+  nbr_axioms := !nbr_axioms + 1;
+  "Why3axiom_" ^ (string_of_int !nbr_axioms)
+
+(* Builtins configuration for propositional logic. *)
+type prop_config =
+  { symb_P     : sym (** Encoding of propositions.        *)
+  ; symb_T     : sym (** Encoding of types.               *)
+  ; symb_or    : sym (** Disjunction(∨) symbol.           *)
+  ; symb_and   : sym (** Conjunction(∧) symbol.           *)
+  ; symb_imp   : sym (** Implication(⇒) symbol.           *)
+  ; symb_bot   : sym (** Bot(⊥) symbol.                   *)
+  ; symb_top   : sym (** Top(⊤) symbol.                   *)
+  ; symb_not   : sym (** Not(¬) symbol.                   *) }
+
+(** [get_prop_config pos builtins] set the builtins configuration using
+  [builtins]. *)
+let get_prop_config :
+  Pos.popt -> sym StrMap.t -> prop_config = fun pos builtins ->
+  let find_sym key =
+    try StrMap.find key builtins with Not_found ->
+    Console.fatal pos "Builtin symbol [%s] undefined." key
+  in
+  { symb_P     = find_sym "P"
+  ; symb_T     = find_sym "T"
+  ; symb_or    = find_sym "or"
+  ; symb_and   = find_sym "and"
+  ; symb_imp   = find_sym "imp"
+  ; symb_bot   = find_sym "bot"
+  ; symb_top   = find_sym "top"
+  ; symb_not   = find_sym "not" }
+
+(** [translate pos builtins (hs, g)] translates from lambdapi to Why3 goal [g]
+  using the hypothesis [hs]. The function return
+  [constants_table, hypothesis, formula] where :
+  - [constants_table] maps abstracted Lambdapi terms to Why3 constants.
+  - [hypothesis] maps abstracted labels to Why3 terms (presentation of
+  [hs]).
+  - [formula] Why3 term representing the goal [g].  *)
+
```

Comment:
No empty line here, otherwise the comment is not attached.

Reply:
Fixed in 9c9ee87dd10ee595ed4af06bce1cc.